src/memory.rs

```diff
@@ -717,6 +723,8 @@ impl Drop for Pipe {
     }
 }
 
+unsafe impl Send for Pipe {}
```

Comment:
`Pipe`s are just created to pass data between running `Kernel`s on a device, it should not have `Send`.